### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by JuanLPalacios**
### Implemented features
-  Create model classes User, Comment, Post and Like.
-  Set up associations between models.
- Methods
  -  User
      - **most_recent_posts:** A method that returns the 3 most recent posts for a given user.
  -  Post
      - **update_user_posts_counter:** A method that updates the posts counter for a user.
      - **most_recent_comments:** A method which returns the 5 most recent comments for a given post.
  -  Comment
      - **update_post_comments**_counter: A method that updates the comments counter for a post.
  -  Like
      - **update_post_likes_counter:** A method that updates the likes counter for a post.
    - ApplicationController
      -  **current_user:** A mock for current user that returns firs user found.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def self.update_user_posts_counter(user)
+    user.update(post_counter: Post.where(author_id: user.id).length)
+  end
```
Comment (medaminedev66) [inline app/models/post.rb]: - Please replace `user` in line 7 with `author` and remove the parameter `user` from the update_user_posts_counter method. Please keep in mind that the author has already been defined on line 2.
- A faster way to update the post_counter would be:
```ruby
def update_user_posts_counter
  author.increment!(:posts_counter)
end
```

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def self.update_post_likes_counter(post)
+    post.update(likes_counter: Like.where(post_id: post.id).length)
+  end
```
Comment (medaminedev66) [inline app/models/like.rb]: - Please remove the parameter `post` from the update_post_likes_counter method. You already defined the `post` in line 3.
- A faster way to update the likes_counter would be:
```ruby
def update_likes_counter
  post.increment!(:likes_counter)
end
```

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def self.update_post_comments_counter(post)
+    post.update(comments_counter: Comment.where(post_id: post.id).length)
+  end
```
Comment (medaminedev66) [inline app/models/comment.rb]: - Please remove the parameter `post` from the update_post_likes_counter method. You already defined the `post` in line 3.
- A faster way to update the comment_counter would be:
```ruby
def update_post_comments_counter
  post.increment!(:comments_counter)
end
```

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @JuanLPalacios  😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- There are no linter errors 
- You have a descriptive PR 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @JuanLPalacios, 

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**